controlplane/kubeadm/internal/workload_cluster_coredns.go

```diff
@@ -125,13 +126,18 @@ func (w *Workload) UpdateCoreDNS(ctx context.Context, kcp *controlplanev1.Kubead
 		return err
 	}
 
-	// Return early if the from/to image is the same.
-	if info.FromImage == info.ToImage {
+	requiresClusterRoleUpdate, err := validateCoreDNSClusterRoleUpdate(version, info)
+	if err != nil {
+		return errors.Wrapf(err, "failed to validate ClusterRole")
+	}
+
+	// Return early if the from/to image is the same and ClusterRole is up to date.
+	if info.FromImage == info.ToImage && !requiresClusterRoleUpdate {
 		return nil
 	}
 
 	// Validate the image tag.
-	if err := validateCoreDNSImageTag(info.FromImageTag, info.ToImageTag); err != nil {
+	if err := validateCoreDNSImageTag(info.FromImageTag, info.ToImageTag, requiresClusterRoleUpdate); err != nil {
```

Comment:
Can we maybe only call this func if .FromImage != ToImage? This should avoid making the logic in the func even messier :)

Reply:
Overall I wonder if we can structure this a bit cleaner.
I.e. call certain stuff if FromImage != ToImage and call certain stuff if requiresClusterRoleUpdate (or maybe some of them if one or the other is true)